internal/core/src/query/visitors/ExecExprVisitor.cpp

```diff
@@ -1782,11 +1788,32 @@ ExecExprVisitor::ExecTermVisitorImplTemplate(TermExpr& expr_raw) -> BitsetType {
     auto index_func = [&terms, n](Index* index) {
         return index->In(n, terms.data());
     };
-    auto elem_func = [&terms, &term_set](MayConstRef<T> x) {
-        //// terms has already been sorted.
-        // return std::binary_search(terms.begin(), terms.end(), x);
+
+// Now only support x86 cpu.
+#if defined(USE_DYNAMIC_SIMD) && defined(__x86_64__)
```

Comment:
Maybe `defined(__x86_64__)` can be controlled inside the simd implementation. In this way, there is only an entry to decide the code path, otherwise we will change everywhere controlled by the `defined(__x86_64__)` after we support new architecture.

Reply:
yes, It can be remove actually, because simd library has ability to judge it.